src/io/iohk/scalanet/peergroup/SimplePeerGroup.scala

```diff
@@ -0,0 +1,113 @@
+package io.iohk.scalanet.peergroup
+
+import java.nio.ByteBuffer
+import java.util.concurrent.ConcurrentHashMap
+
+import io.iohk.scalanet.messagestream.MessageStream
+import io.iohk.scalanet.peergroup.PeerGroup.{Lift, NonTerminalPeerGroup}
+import io.iohk.scalanet.peergroup.SimplePeerGroup.Config
+
+import scala.collection.mutable
+import scala.language.higherKinds
+import scala.collection.JavaConverters._
+import io.iohk.decco.auto._
+import io.iohk.decco._
+import SimplePeerGroup._
+import monix.eval.Task
+import org.slf4j.LoggerFactory
+
+class SimplePeerGroup[A, F[_], AA](
```

Comment:
Try to use specific type names instead, I have no idea what does `A` nor `AA` means.